src/lib/parameters/parameters.cpp

```diff
@@ -584,19 +546,42 @@ param_get(param_t param, void *val)
 int
 param_get_default_value(param_t param, void *default_val)
 {
-	if (default_val && handle_in_range(param)) {
+	if (!handle_in_range(param)) {
+		PX4_ERR("get default value: param %d invalid", param);
+		return PX4_ERROR;
+	}
+
+	if (default_val) {
+		if (params_custom_default[param] && param_custom_default_values) {
+			// get default from custom default storage
+			param_lock_custom_defaults();
```

Comment:
We're now using this lock when we already have `param_lock_writer`. Do we really need both locks or could we have just one for both? I'm asking because I'd be scared of deadlocks.

Reply:
Let me go through it again and see if we can do without.

Reply:
Refactored a bit to share the existing reader/writer locking.